testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/CLIEmbedHostControllerTestCase.java

```diff
@@ -472,6 +480,294 @@ public void testPrivateHiddenOperations() throws IOException {
+    }
+
+    @Test
+    public void testEmptyHostConfigWithAddAndReloadOnHost() throws Exception {
+        final String line = "embed-host-controller --std-out=echo --host-config=host-empty-cli.xml --empty-host-config --remove-existing-host-config " + JBOSS_HOME;
+        cli.sendLine(line);
+        assertTrue(cli.isConnected());
+        cli.sendLine("/host=foo:add()");
+        // host-empty-cli.xml
+        assertTrue(cli.isConnected());
+        cli.sendLine("/host=foo/core-service=management/security-realm=test-realm:add()");
+        cli.sendLine("/host=foo/core-service=management/security-realm=test-realm/authentication=local:add(default-user=\"$local\", skip-group-loading=true)");
+        cli.sendLine("/host=foo/interface=test-mgmt-interface:add(inet-address=${jboss.bind.address.management:127.0.0.1})");
+        cli.sendLine("/host=foo/core-service=management/management-interface=http-interface:add(interface=test-mgmt-interface, security-realm=test-realm, http-upgrade-enabled=true, port=${jboss.management.http.port:9990})");
+        // check things are still there after reload:
+        cli.sendLine("reload --host=foo --admin-only=true");
+        assertTrue(cli.isConnected());
+        cli.sendLine("ls /host=foo/core-service=management/security-realm=test-realm");
+        cli.sendLine("ls /host=foo/interface=test-mgmt-interface");
+        cli.sendLine("ls /host=foo/core-service=management/management-interface=http-interface");
+    }
+
+    @Test
+    public void addNewDomainController() throws Exception {
+        final String line = "embed-host-controller --std-out=echo --host-config=host-empty-cli.xml --empty-host-config --remove-existing-host-config " + JBOSS_HOME;
+        cli.sendLine(line);
+        assertTrue(cli.isConnected());
+        cli.sendLine("/host=foomaster:add()");
+        // host-empty-cli.xml
+        assertTrue(cli.isConnected());
+        cli.sendLine("/host=foomaster/core-service=management/security-realm=test-realm:add()");
+        cli.sendLine("/host=foomaster/core-service=management/security-realm=test-realm/authentication=local:add(default-user=\"$local\", skip-group-loading=true)");
+        cli.sendLine("/host=foomaster/interface=test-mgmt-interface:add(inet-address=${jboss.bind.address.management:127.0.0.1})");
+        cli.sendLine("/host=foomaster/core-service=management/management-interface=http-interface:add(interface=test-mgmt-interface, security-realm=test-realm, http-upgrade-enabled=true, port=${jboss.management.http.port:9990})");
+        // check things are still there after reload:
+        cli.sendLine("reload --host=foomaster --admin-only=true");
+        assertTrue(cli.isConnected());
+        cli.sendLine("ls /host=foomaster/core-service=management/security-realm=test-realm");
+        cli.sendLine("ls /host=foomaster/interface=test-mgmt-interface");
+        cli.sendLine("ls /host=foomaster/core-service=management/management-interface=http-interface");
+    }
```

Comment:
Is L666 through here testing different stuff from what came above it? It looks the same. If it's different can you add code comments to make the difference more apparent?

Reply:
I've cleaned these up and just kept the one that adds a slave.